src/app/shared/menu/date/custom-date-range-dropdown.component.ts

```diff
@@ -1,48 +1,87 @@
-import { FilterModel, INTERVAL_HOURLY } from '../../../ngrx';
+import { FilterModel, INTERVAL_HOURLY, IntervalModel, IntervalList } from '../../../ngrx';
 import * as dateUtil from '../../util/date';
 import { GoogleAnalyticsEventAction } from '../../../ngrx/actions';
 
 @Component({
   selector: 'metrics-custom-date-range-dropdown',
   template: `
-    <div class="custom-date-range-dropdown" [class.open]="open">
+    <div class="dropdown" [class.open]="open">
       <div class="overlay" (click)="toggleOpen()"></div>
-      <div class="custom-date-range-button">
+      <div class="dropdown-button">
         <button class="btn-icon icon-calendar grey-dove" (click)="toggleOpen()" aria-label="Custom Date Range"></button>
       </div>
-      <div class="custom-date-range-content">
-        <prx-daterange [from]="dateRange.beginDate" [to]="dateRange.endDate" UTC="true"
-                       (rangeChange)="onCustomRangeChange($event)"></prx-daterange>
-        <div class="invalid" *ngIf="invalid">
-          {{ invalid }}
+      <div class="dropdown-content">
+        <div class="intervals">
+          <button *ngFor="let interval of intervalList"
+                  [class.btn-link]="tempFilter?.interval !== interval"
+                  (click)="onIntervalChange(interval)">{{ interval.name }}
+          </button>
+        </div>
+        <hr>
+        <div class="ranges">
+          <div class="pickers">
+            <prx-daterange [from]="tempFilter.beginDate" [to]="tempFilter.endDate" UTC="true"
+                           (rangeChange)="onCustomRangeChange($event)"></prx-daterange>
+            <div class="invalid" *ngIf="invalid">
+              {{ invalid }}
+            </div>
+          </div>
+          <div class="separator"></div>
+          <metrics-standard-date-range [standardRange]="tempFilter?.standardRange" [interval]="tempFilter?.interval"
+                                       (standardRangeChange)="onStandardRangeChange($event)">
+          </metrics-standard-date-range>
         </div>
         <p class="buttons">
           <button (click)="toggleOpen()" class="btn-link">Cancel</button>
-          <button (click)="onApply()">Apply</button>
+          <button (click)="onApply()" [disabled]="invalid">Apply</button>
         </p>
       </div>
     </div>
   `,
-  styleUrls: ['./custom-date-range-dropdown.component.css']
+  styleUrls: ['../dropdown.css', './custom-date-range-dropdown.component.css']
 })
 
 export class CustomDateRangeDropdownComponent implements OnChanges {
   @Input() filter: FilterModel;
   @Output() dateRangeChange = new EventEmitter<FilterModel>();
-  dateRange: FilterModel;
+  tempFilter: FilterModel;
   open = false;
+  intervalList = IntervalList;
+  userChoseRange: string;
+  CUSTOM_DATE = 'custom-date';
+  STANDARD_DATE = 'standard-date';
 
   constructor(public store: Store<any>) {}
 
   ngOnChanges() {
-    this.dateRange = this.filter;
+    this.tempFilter = this.filter;
```

Comment:
Not sure I understand why you copy `this.filter -> this.tempFilter` both here and in the toggleOpen.  I guess I thought maybe just toggleOpen would be enough?

Reply:
You're right, it didn't need this onChanges. Fixed.